mitmproxy/addons/modifybody.py

```diff
@@ -0,0 +1,98 @@
+import os
+import re
+import typing
+
+from mitmproxy import exceptions
+from mitmproxy import flowfilter
+from mitmproxy import ctx
+from mitmproxy.addons.modifyheaders import parse_modify_hook
+
+
+class ModifyBody:
+    def __init__(self):
+        self.lst = []
+
+    def load(self, loader):
+        loader.add_option(
+            "modify_body", typing.Sequence[str], [],
+            """
+            Replacement pattern of the form "[/flow-filter]/regex/[@]replacement", where
+            the separator can be any character. The @ allows to provide a file path that
+            is used to read the replacement string.
+            """
+        )
+
+    def configure(self, updated):
+        """
+            .modify_body is a list of tuples (flow_filter_pattern, regex, repl):
+
+            flow_filter_pattern: a string specifying a flow filter pattern.
+            regex: a regular expression, as string.
+            repl: the replacement string
+        """
+        if "modify_body" in updated:
+            lst = []
+            for rep in ctx.options.modify_body:
+                try:
+                    flow_filter_pattern, regex, repl = parse_modify_hook(rep)
+                except ValueError as e:
+                    raise exceptions.OptionsError(
+                        "Invalid modify_body option: %s" % rep
```

Comment:
Make this `f"Cannot parse modify_body option {rep}: {e}"` to include the ValueError's message, then move the checks below into `parse_modify_hook` and make their messages generic (`"Invalid modify_body flow filter: %s" % flow_filter_pattern` -> `Invalid filter pattern: {pattern}`). This should solve the problem mentioned in https://github.com/mitmproxy/mitmproxy/pull/4046#discussion_r447724841, or am I misunderstanding things? :)

Reply:
Yep I guess this makes sense :)